### PR TITLE
Robodoc: Adding Robodoc documentation tool

### PR DESCRIPTION
**Pull request by wscullin**
Contributing the Robodoc software documentation tool.

### CODE REVIEW

File: var/spack/repos/builtin/packages/robodoc/package.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Robodoc(AutotoolsPackage):
+    """ROBODoc is program documentation tool."""
+
+    homepage = "http://www.xs4all.nl/~rfsber/Robo/index.html"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/robodoc/package.py]:
```suggestion
    homepage = "https://www.xs4all.nl/~rfsber/Robo/index.html"
```
Or perhaps use the URL this redirects to (i.e., https://rfsber.home.xs4all.nl/Robo/index.html)?

**Review by tldahlgren — CHANGES_REQUESTED** [review]
We prefer the use of `https` over `http` for all urls (when possible).

File: var/spack/repos/builtin/packages/robodoc/package.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class Robodoc(AutotoolsPackage):
+    """ROBODoc is program documentation tool."""
+
+    homepage = "https://www.xs4all.nl/~rfsber/Robo/index.html"
+    url      = "https://github.com/gumpu/ROBODoc/archive/refs/tags/v4.99.44.tar.gz"
+
+    maintainers = ['wscullin']
+
+    version('develop', git='https://github.com/gumpu/ROBODoc.git')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/robodoc/package.py:18]: I would add `git = ...` above where you define the url. Then you can use `branch='release'` here. This way, if someone ever adds master, you don't need to duplicate the git.

File: var/spack/repos/builtin/packages/robodoc/package.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    version('4.99.44', sha256='8ed875bbde2788d7bc986693077577d6cc6e15e4bc660d522164710977952e90')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    def autoreconf(self, spec, prefix):
+        autoreconf('--install', '--verbose', '--force')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/robodoc/package.py:27]: Does the default implementation not work?

File: var/spack/repos/builtin/packages/robodoc/package.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    def configure_args(self):
+        args = []
+        return args
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/robodoc/package.py:31]: You can remove this if you aren't going to use it.

**Comment by tldahlgren** [comment]
Is this PR still active?  Adam made a number of useful suggestions.

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**